### ISSUE TITLE
Terrablender fails to generate biomes

### ISSUE THREAD

**Issue by Yucked** [DenisMasterHerobrine/MagicalForest#3]
When using this mod with other biome related mods, it seems terrablender gets confused with biome names.

```java
[13:13:14] [main/FATAL]: Failed to start the minecraft server
java.lang.IllegalStateException: Feature order cycle found, involved biomes: [net.minecraft.class_1959@235dd97b, net.minecraft.class_1959@9e60054, net.minecraft.class_1959@e878155, net.minecraft.class_1959@2e430cc6]
	at net.minecraft.class_1966.method_39525(class_1966.java:130) ~[server-intermediary.jar:?]
	at net.minecraft.class_1966.<init>(class_1966.java:65) ~[server-intermediary.jar:?]
	at net.minecraft.class_1966.<init>(class_1966.java:58) ~[server-intermediary.jar:?]
	at terrablender.worldgen.TBMultiNoiseBiomeSource.<init>(TBMultiNoiseBiomeSource.java:74) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
	at terrablender.worldgen.TBMultiNoiseBiomeSource$Preset.biomeSource(TBMultiNoiseBiomeSource.java:144) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
	at terrablender.worldgen.TBMultiNoiseBiomeSource$Preset.biomeSource(TBMultiNoiseBiomeSource.java:149) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
	at terrablender.api.WorldPresetUtils.chunkGenerator(WorldPresetUtils.java:110) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
	at terrablender.api.WorldPresetUtils.overworldChunkGenerator(WorldPresetUtils.java:48) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
	at net.minecraft.class_5285.handler$doa000$onCreate(class_5285.java:567) ~[server-intermediary.jar:?]
	at net.minecraft.class_5285.method_28021(class_5285.java:232) ~[server-intermediary.jar:?]
	at net.minecraft.class_3806.method_37371(class_3806.java:92) ~[server-intermediary.jar:?]
	at net.minecraft.server.Main.main(Main.java:173) [server-intermediary.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608) [fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.12.12.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:66) [fabric-loader-0.12.12.jar:?]
```

**Comment by DenisMasterHerobrine** [comment]
Which mods do you use exactly to get this crash? Also is this happening only serverside or the same is caused when creating a world in singleplayer? Seems like Terrablender got crashed because shuffled biome features from all the biomes.
You can also try to put [Cyanide for Fabric](https://www.curseforge.com/minecraft/mc-mods/cyanide-fabric) into your modpack or whatever it is to get a better logged error to find out what caused it and send it here.

**Comment by Chuunkyz** [comment]
same problem here

**Comment by Chuunkyz** [comment]
Error(s) loading registry minecraft:worldgen/biome:
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/nether_jungle.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/flooded_deltas.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/crimson_glowing_woods.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/sulfuric_bone_reef.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/gravel_desert.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/upside_down_forest.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/nether_grasslands.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/bone_reef.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/wart_forest_edge.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/crimson_pinewood.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/wart_forest.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/nether_mushroom_forest.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/poor_nether_grasslands.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/nether_mushroom_forest_edge.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/soul_plain.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/upside_down_forest_cleared.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/old_warped_woods.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/magma_land.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/old_swampland.json"
Unknown element: ResourceKey[minecraft:worldgen/placed_feature / betternether:nether_ruby_rare_ore]
	at: file "data/betternether/worldgen/placed_feature/nether_ruby_rare_ore.json"
	at: reference to "betternether:nether_ruby_rare_ore" from minecraft:worldgen/placed_feature
	at: "features", step underground_ores, index 6
	at: file "data/betternether/worldgen/biome/old_fungiwoods.json"

also, nether ruby from better is **disabled** on my instance

**Comment by Yucked** [comment]
> Which mods do you use exactly to get this crash? Also is this happening only serverside or the same is caused when creating a world in singleplayer? Seems like Terrablender got crashed because shuffled biome features from all the biomes. You can also try to put [Cyanide for Fabric](https://www.curseforge.com/minecraft/mc-mods/cyanide-fabric) into your modpack or whatever it is to get a better logged error to find out what caused it and send it here.

I've about 164 mods loaded so narrowing it down is going be slightly difficult but when adding your mod, it throws that exception but to name few:
- All the biomes you'll go
- Amplifier Nether
- Atlantis
- Azalea
- Graves
- Lush Forest
- More Geodes
- Promenade
- Puddles
- The Graveyard

There might be few that I may have missed and this is server-sided, haven't tried it on single-player world generation. Using cyanide, I get the following:

```java
[14:32:58] [main/FATAL]: Failed to start the minecraft server
com.alcatrazescapee.cyanide.codec.FeatureCycleDetector$FeatureCycleException: A feature cycle was found.

Cycle:
At step 9
Feature 'minecraft:trees_water'
  must be before 'minecraft:flower_default' (defined in 'minecraft:deep_cold_ocean' at index 1, 2 and 10 others)
  must be before 'minecraft:patch_grass_badlands' (defined in 'minecraft:deep_frozen_ocean' at index 2, 3 and 21 others)
  must be before 'minecraft:trees_savanna' (defined in 'byg:araucaria_savanna' at index 1, 2)
  must be before 'minecraft:flower_warm' (defined in 'byg:baobab_savanna' at index 2, 3 and 3 others)
  must be before 'minecraft:trees_water' (defined in 'magicalforest:magical_forest' at index 2, 3)

        at com.alcatrazescapee.cyanide.codec.FeatureCycleDetector.buildFeaturesPerStep(FeatureCycleDetector.java:150) ~[cyanide-fabric-1.18-pre6-2.0.2.jar:?]
        at com.alcatrazescapee.cyanide.codec.MixinHooks.buildFeaturesPerStepAndPopulateErrors(MixinHooks.java:70) ~[cyanide-fabric-1.18-pre6-2.0.2.jar:?]
        at net.minecraft.class_1966.handler$bkd000$buildFeaturesPerStepWithAdvancedCycleDetection(class_1966.java:1028) ~[server-intermediary.jar:?]
        at net.minecraft.class_1966.method_39525(class_1966.java) ~[server-intermediary.jar:?]
        at net.minecraft.class_1966.<init>(class_1966.java:65) ~[server-intermediary.jar:?]
        at net.minecraft.class_1966.<init>(class_1966.java:58) ~[server-intermediary.jar:?]
        at terrablender.worldgen.TBMultiNoiseBiomeSource.<init>(TBMultiNoiseBiomeSource.java:74) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
        at terrablender.worldgen.TBMultiNoiseBiomeSource$Preset.biomeSource(TBMultiNoiseBiomeSource.java:144) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
        at terrablender.worldgen.TBMultiNoiseBiomeSource$Preset.biomeSource(TBMultiNoiseBiomeSource.java:149) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
        at terrablender.api.WorldPresetUtils.chunkGenerator(WorldPresetUtils.java:110) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
        at terrablender.api.WorldPresetUtils.overworldChunkGenerator(WorldPresetUtils.java:48) ~[TerraBlender-fabric-1.18.1-1.1.0.73.jar:?]
        at net.minecraft.class_5285.handler$eig000$onCreate(class_5285.java:567) ~[server-intermediary.jar:?]
        at net.minecraft.class_5285.method_28021(class_5285.java:232) ~[server-intermediary.jar:?]
        at net.minecraft.class_3806.method_37371(class_3806.java:92) ~[server-intermediary.jar:?]
        at net.minecraft.server.Main.main(Main.java:173) [server-intermediary.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608) [fabric-loader-0.12.12.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.12.12.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.12.12.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:66) [fabric-loader-0.12.12.jar:?]
```

**Comment by Chuunkyz** [comment]
my issue is related to better nether btw...

**Comment by DenisMasterHerobrine** [comment]
Fixed in 2.0.1 I believe. Will test Magical Forest with other biome mods a bit more in case I find other incompatibility, then I'll release it on CurseForge soon.

**Comment by DenisMasterHerobrine** [comment]
Check out the 2.0.1 version on CurseForge. Mod should load without any problems now. However if something breaks again when loading a world - reopen an issue.
